pallets/ocex/src/lib.rs

```diff
@@ -667,27 +671,30 @@ pub mod pallet {
 			snapshot_id: u32,
 			beneficiary: T::AccountId,
 		) -> DispatchResult {
-			// TODO: The caller should be of operational council
 			T::GovernanceOrigin::ensure_origin(origin)?;
-
-			let fees: Vec<Fees> = <FeesCollected<T>>::get(snapshot_id).iter().cloned().collect();
-			for fee in fees {
-				if let Some(converted_fee) =
-					fee.amount.saturating_mul(Decimal::from(UNIT_BALANCE)).to_u128()
-				{
-					Self::transfer_asset(
-						&Self::get_custodian_account(),
-						&beneficiary,
-						converted_fee.saturated_into(),
-						fee.asset,
-					)?;
-				// TODO: Remove the fees from storage if successful
-				} else {
-					return Err(Error::<T>::FailedToConvertDecimaltoBalance.into())
+			<FeesCollected<T>>::mutate(snapshot_id, |fees| {
+				let mut fee_claim_rounds = T::SnapshotFeeClaim::get();
+				while let Some(fee) = fees.pop() {
+					if let Some(converted_fee) =
+						fee.amount.saturating_mul(Decimal::from(UNIT_BALANCE)).to_u128()
+					{
+						Self::transfer_asset(
+							&Self::get_custodian_account(),
+							&beneficiary,
+							converted_fee.saturated_into(),
+							fee.asset,
+						)?;
+					} else {
+						return Err(Error::<T>::FailedToConvertDecimaltoBalance.into())
```

Comment:
Didn't we need to put fee item back in collection since we are doing 'pop' on 677 line?

Reply:
So by the @zktony comment - this is intended. Then different question - do we really want to drop this fee charge? I mean - mby we should define some minimum fee amount and charge at least this? or some average value?